wavedef.py

```diff
@@ -131,52 +136,113 @@ def __init__(self, name, mappings):
     self.mappings = mappings
 # pylint: enable=too-few-public-methods
 
-# format
-_WAVEFORM_DEFS = [
-    WaveformDef('sin', {':': 0}),
-    WaveformDef('square', {':': 1}),
-    WaveformDef('cmos', {':': 2}),
-    WaveformDef('adj-pulse', {':0': 3}),
-    WaveformDef('dc', {':0': 4, ':1': 3}),
-    WaveformDef('tri', {':0': 5, ':1': 4}),
-    WaveformDef('ramp', {':0': 6, ':1': 5}),
-    WaveformDef('neg-ramp', {':0': 7, ':1': 6}),
-    WaveformDef('stair-tri', {':0': 8, ':1': 7}),
-    WaveformDef('stair', {':0': 9, ':1': 8}),
-    WaveformDef('neg-stair', {':0': 10, ':1': 9}),
-    WaveformDef('exp', {':0': 11, ':1': 10}),
-    WaveformDef('neg-exp', {':0': 12, ':1': 11}),
-    WaveformDef('fall-exp', {':0': 13, ':1': 12}),
-    WaveformDef('neg-fall-exp', {':0': 14, ':1': 13}),
-    WaveformDef('log', {':0': 15, ':1': 14}),
-    WaveformDef('neg-log', {':0': 16, ':1': 15}),
-    WaveformDef('fall-log', {':0': 17, ':1': 16}),
-    WaveformDef('neg-fall-log', {':0': 18, ':1': 17}),
-    WaveformDef('full-wav', {':0': 19, ':1': 18}),
-    WaveformDef('neg-full-wav', {':0': 20, ':1': 19}),
-    WaveformDef('half-wav', {':0': 21, ':1': 20}),
-    WaveformDef('neg-half-wav', {':0': 22, ':1': 21}),
-    WaveformDef('lorentz', {':0': 23, ':1': 22}),
-    WaveformDef('multitone', {':0': 24, ':1': 23}),
-    WaveformDef('rand', {':0': 25, ':1': 24}),
-    WaveformDef('ecg', {':0': 26, ':1': 25}),
-    WaveformDef('trap', {':0': 27, ':1': 26}),
-    WaveformDef('sinc', {':0': 28, ':1': 27}),
-    WaveformDef('impulse', {':0': 29, ':1': 28}),
-    WaveformDef('gauss', {':0': 30, ':1': 29}),
-    WaveformDef('am', {':0': 31, ':1': 30}),
-    WaveformDef('fm', {':0': 32, ':1': 31}),
-    WaveformDef('chirp', {':0': 33, ':1': 32}),
-]
-
 def _make_arb(count, start_dict):
+  result = {}
   for arb_index in range(1, count + 1):
-    _WAVEFORM_DEFS.append(WaveformDef('arb%u' % arb_index, start_dict))
-    # create a new dictionary will all indexes incremented by one
-    start_dict = dict((k, v+1) for k, v in six.iteritems(start_dict))
+    result[f'arb{arb_index}'] = start_dict
+    start_dict = dict((k, v + 1) for k, v in six.iteritems(start_dict))
+  return result
+
+_WAVEFORMS_2300_6600_6800 = {
+    'sin': {':': 0},
+    'square': {':': 1},
+    'cmos': {':': 2},
+    'adj-pulse': {':0': 3},
+    'dc': {':0': 4, ':1': 3},
+    'tri': {':0': 5, ':1': 4},
+    'ramp': {':0': 6, ':1': 5},
+    'neg-ramp': {':0': 7, ':1': 6},
+    'stair-tri': {':0': 8, ':1': 7},
+    'stair': {':0': 9, ':1': 8},
+    'neg-stair': {':0': 10, ':1': 9},
+    'exp': {':0': 11, ':1': 10},
+    'neg-exp': {':0': 12, ':1': 11},
+    'fall-exp': {':0': 13, ':1': 12},
+    'neg-fall-exp': {':0': 14, ':1': 13},
+    'log': {':0': 15, ':1': 14},
```

Comment:
I think that a single dictionary is less complex than 2 dictionaries with combination logic.  e.g.

'log': {'f6900:0': 17, 'f6900:1': 16, ':0': 15, ':1': 14},

Reply:
Fair enough! Changed to one dictionary

Reply:
Thanks!